app/Makefile

```diff
@@ -6,11 +6,13 @@ CURRENT_UID              := $(shell id -u)
 CURRENT_GID              := $(shell id -g)
 MKFILE_PATH              := $(abspath $(lastword $(MAKEFILE_LIST)))
 CURRENT_DIR              := $(patsubst %/,%,$(dir $(MKFILE_PATH)))
-
 # By default use docker volume to persist mvn cache,
 # but allow passing a directory, e.g., local ~/.m2
 mvn_cache_docker_volume := mvn-cache-up4
 MVN_CACHE ?= ${mvn_cache_docker_volume}
+# By default use the maven settings in local ~/.m2, but allow passing a custom
+# settings.
+MVN_SETTINGS ?= $(shell test -f ~/.m2/settings.xml && echo "$${HOME}/.m2/settings.xml")
```

Comment:
Just curious - is there any specific reason why this can't be part of the `mvn_settings` section below? This variable doesn't seem used anywhere else

Reply:
The reason is that I'm using `MVN_SETTINGS` in an `ifneq` in the `mvn_settings` target. For this reason, I can't use eval in the make target to set the variable. 
Also, defining it here would allow user to provide their custom maven settings file by overriding the `MVN_SETTINGS` make variable.
Hope I have got your comment right. Please, let me know if you had something different in mind.